tests/Unit/Domain/ShoppingItemStackTest.php

```diff
@@ -53,6 +53,23 @@ public function testPush(): ShoppingItemStack
         return $actual;
     }
 
+    public function testRemove(): ShoppingItemStack
+    {
+        $expected = [
+                new ShoppingItem(1, 'Peaches'),
+                new ShoppingItem(2, 'Bananas'),
+                new ShoppingItem(3, 'Apples'),
+            ];
+
+        $stack = new ShoppingItemStack($expected[0], $expected[2], $expected[1]);
+        $actual = $stack->remove($expected[2]);
+
+        $this->assertCount(2, $actual);
+        $this->assertSame([$expected[0], $expected[1]], $actual->all());
+
+        return $actual;
```

Comment:
Comment: test here looks great, well done on writing it. The only thing you don't need to do is return the stack at the end of the test. You only need to return a value from one test if it then needs to be used in a subsequent test. That's indicated by the `@depends` annotation.

When removing the return from this test, you'd also need to change the return type-hint to `void`.

Reply:
Ah, okay makes sense. I'm just testing if remove works and if it does nothing would be returned.